lib/Rex/Config.pm

```diff
@@ -682,6 +682,9 @@ sub get_ssh_config_hostname {
     if ( $SSH_CONFIG_FOR{ $param->{server} }->{hostname} =~ m/^\%h\.(.*)/ ) {
       return $param->{server} . "." . $1;
     }
+    elsif ( $SSH_CONFIG_FOR{ $param->{server} }->{hostname} =~ m/^\%h$/ ) {
```

Comment:
THis works. This could be a literal match to "%h". Heck, you could match this line OR "%h". -- m!

Reply:
Overengineering is the name of the game (or in this case, copy pasting the previous regex instead of thinking about things for 2s). Will simplify and then merge, ta! :)

Reply:
Speaking about overengineering, you could also have a single if clause with the regex `^\%h(\.(.*))?$` and append `$1` without the additional dot. I'm fine either way.

Could you be so kind to submit this to [RexOps/Rex](https://github.com/RexOps/Rex) as well? This is potentially useful for others as well.

Reply:
Thank you, that's a better idea. I'll look after upstreaming after we merge it here